src/parallel/executor.cpp

```diff
@@ -426,18 +429,34 @@ void Executor::WorkOnTasks() {
 	}
 }
 
+void Executor::SignalTaskRescheduled(lock_guard<mutex> &) {
+	task_reschedule.notify_one();
+}
+
+void Executor::WaitForTask() {
+	std::unique_lock<mutex> executor_guard(executor_lock);
+	if (to_be_rescheduled_tasks.empty()) {
+		return;
+	}
+	std::unique_lock<mutex> reschedule_guard(task_reschedule_lock);
+	executor_guard.unlock();
+	task_reschedule.wait(reschedule_guard);
```

Comment:
Can we pass in the `executor_guard` here, and remove the `task_reschedule_lock` in its entirety? Or would that not work?

Reply:
It might actually, I think I was worried about using an existing lock for the `condition_variable`, thinking we would want to keep the `executor_lock` held, but to fix the race condition I have to unlock the `executor_lock` anyways

I'll make the change and report the findings, I suspect this is the right solution 👍